linkerd/http-route/src/grpc/match.rs

```diff
@@ -0,0 +1,104 @@
+#[cfg(test)]
+mod tests;
+
+use crate::http::MatchHeader;
+
+/// Matches gRPC routes.
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+pub struct MatchRoute {
+    pub(crate) rpc: MatchRpc,
+    pub(crate) headers: Vec<MatchHeader>,
+}
+
+/// Summarizes a matched gRPC route.
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+pub struct RouteMatch {
+    rpc: RpcMatch,
+    headers: usize,
+}
+
+/// Matches gRPC endpoints.
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct MatchRpc {
+    pub(crate) service: Option<String>,
+    pub(crate) method: Option<String>,
+}
+
+/// Summarizes a matched gRPC endpoints.
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct RpcMatch {
+    service: usize,
+    method: usize,
```

Comment:
IMO it would be nice if it was documented somewhere that the match results are represented as the length of the prefix that was matched?